dsf-bpe/dsf-bpe-process-feasibility/src/main/java/org/highmed/dsf/bpe/service/DownloadResearchStudyResource.java

```diff
@@ -73,27 +88,50 @@ private ResearchStudy getResearchStudy(IdType researchStudyid, FhirWebserviceCli
 		}
 		catch (WebApplicationException e)
 		{
-			throw new ResourceNotFoundException("Error while reading ResearchStudy with id "
-					+ researchStudyid.getIdPart() + " from " + client.getBaseUrl());
+			throw new ResourceNotFoundException(
+					"Error while reading ResearchStudy with id " + researchStudyid.getIdPart() + " from " + client
+							.getBaseUrl());
 		}
 	}
 
+	private ResearchStudy addMissingOrganizations(ResearchStudy researchStudy, FhirWebserviceClient client)
+	{
+		List<String> medicReferences = organizationProvider.getOrganizationsByType("MeDIC").map(organization -> {
+			IdType type = IdType.of(organization);
+			return type.getResourceType() + "/" + type.getIdPart();
+		}).collect(Collectors.toList());
+
+		List<String> targetReferences = researchStudy.getExtension().stream()
+				.filter(extension -> extension.getUrl().equals(Constants.EXTENSION_PARTICIPATING_MEDIC_URI))
+				.map(extension -> ((Reference) extension.getValue()).getReference()).collect(Collectors.toList());
+
+		medicReferences.forEach(reference -> {
+			if (!targetReferences.contains(reference))
+			{
+				Extension extension = new Extension(Constants.EXTENSION_PARTICIPATING_MEDIC_URI,
+						new Reference(reference));
+				researchStudy.addExtension(extension);
```

Comment:
add warning log entry, something like: "adding missing MeDIC organization ... to research study with id ..."